src/Routing/Middleware/AssetMiddleware.php

```diff
@@ -98,13 +74,15 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
             return $next($request, $response);
         }
 
-        $file = new File($assetFile);
-        $modifiedTime = $file->lastChange();
+        $file = new SplFileInfo($assetFile);
+        $modifiedTime = $file->getMTime();
         if ($this->isNotModified($request, $file)) {
-            $headers = $response->getHeaders();
-            $headers['Last-Modified'] = date(DATE_RFC850, $modifiedTime);
-
-            return new Response('php://memory', 304, $headers);
+            return $response
+                ->withStatus(304)
+                ->withHeader(
+                    'Last-Modified',
+                    date(DATE_RFC850, $modifiedTime)
+                );
```

Comment:
A 304 response shouldn't have a body. You should overwrite the body here.

Reply:
Added `->withStringBody('')`.